service/history/queues/executable.go

```diff
@@ -383,6 +356,15 @@ func (e *executableImpl) writeToDLQ(ctx context.Context) error {
 }
 
 func (e *executableImpl) isSafeToDropError(err error) bool {
+	if errors.Is(err, consts.ErrStaleReference) {
+		// The task is stale and is safe to be dropped.
+		// Even though ErrStaleReference is castable to serviceerror.NotFound, we give this error special treatment
+		// because we're interested in the metric.
+		metrics.TaskSkipped.With(e.taggedMetricsHandler).Record(1)
```

Comment:
My understanding it that it should never happen and basically the same as the StaleStateError.

> goes into a feature branch

I am fine either way. That dropping logic is there for a long time so not an urgent thing to fix. 
I commented here because it might impact the interpretation of the TaskSkipped metric.